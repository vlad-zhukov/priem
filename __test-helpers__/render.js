/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import {render} from 'react-testing-library';

export default (element, options) => {
    const ref = React.createRef();
    const el = React.cloneElement(element, {ref});
    const result = render(el, options);
    return {...result, instance: ref.current};
};
