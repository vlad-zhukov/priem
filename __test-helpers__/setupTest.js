/* eslint-disable import/no-extraneous-dependencies */

import 'raf/polyfill';
