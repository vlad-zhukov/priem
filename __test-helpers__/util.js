/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import delay from 'delay';
import Priem from '../src/Priem';
import {Container, populateStore} from '../src/Container';

export function testComponent({initialStore, options} = {}) {
    if (initialStore) {
        populateStore(initialStore);
    }

    const ctr = new Container({
        mapPropsToArgs: () => ['foo'],
        promise: value => delay(100, {value}),
        ...options,
    });

    return <Priem sources={{ctr}}>{p => <div>{p.ctr}</div>}</Priem>;
}

export function testComponentNested({initialStore, ctr1Props, ctr2Props} = {}) {
    if (initialStore) {
        populateStore(initialStore);
    }

    const ctr1 = new Container({
        mapPropsToArgs: () => ['foo'],
        promise: value => delay(100, {value}),
        ...ctr1Props,
    });

    const ctr2 = new Container({
        mapPropsToArgs: p => (!p.ctr1 ? null : [p.ctr1, 'bar']),
        promise: (ctr1Value, value) => delay(100, {value: ctr1Value + value}),
        ...ctr2Props,
    });

    return (
        <Priem sources={{ctr1, ctr2}}>
            {props => (
                <Priem sources={{ctr2}} ctr1={props.ctr1}>
                    {p => <div>{p.ctr2}</div>}
                </Priem>
            )}
        </Priem>
    );
}

/* eslint-disable react/no-unused-state */
export class ErrorBoundary extends React.Component {
    constructor() {
        super();
        this.state = {initTime: Date.now(), hasError: null};
    }

    componentDidCatch(error) {
        this.setState({hasError: error, catchTime: Date.now()});
    }

    render() {
        if (this.state.hasError) {
            return null;
        }
        return this.props.children;
    }
}
/* eslint-enable react/no-unused-state */
