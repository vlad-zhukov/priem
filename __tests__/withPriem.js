import React from 'react';
import delay from 'delay';
import {mount} from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import {TestComponentSimpleDecorated, TestComponentNestedDecorated} from '../__test-helpers__/util';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

it('should render simple decorated component', async () => {
    const wrapper = mount(<TestComponentSimpleDecorated />);
    await delay(150);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
});

it('should render nested decorated component', async () => {
    const wrapper = mount(<TestComponentNestedDecorated />);
    await delay(300);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
});
