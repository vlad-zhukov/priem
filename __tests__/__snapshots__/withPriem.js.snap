// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a nested decorated component 1`] = `"<div>foobar</div>"`;

exports[`should render a simple decorated component 1`] = `"<div>foo</div>"`;
